File: AkitaModel/Results/DepartmentResults.cs
using System.Collections.Generic;

namespace Skyware.Lis.AkitaModel.Results;

/// <summary>
/// Wrapper class of examination results that follows structure of the printout.
/// </summary>
public class DepartmentResults
{

    /// <summary>
    /// PK of the Department in LIS iLab.
    /// </summary>
    public int DepartmentId { get; set; }

    /// <summary>
    /// Name of the department.
    /// </summary>
    public string DepartmentName { get; set; } = string.Empty;

    /// <summary>
    /// Rank of the department, use it for sorting.
    /// </summary>
    public int Rank { get; set; }

    /// <summary>
    /// Label for the reference range column for this department.
    /// </summary>
    public string RangeLabel { get; set; }

    /// <summary>
    /// Department-level note.
    /// </summary>
    public Note Note { get; set; }

    /// <summary>
    /// List of examination results.
    /// </summary>
    public IList<ExaminationBase> Results { get; set; }

}

File: AkitaMotelTS/Results/DepartmentResults.ts
import { Note } from "./Note";
import { ExaminationBase } from "./ExaminationBase";

namespace Skyware.Lis.AkitaModel.Results {
  /**
   * Wrapper class of examination results that follows structure of the printout.
   */
  export class DepartmentResults {
    /**
     * PK of the Department in LIS iLab.
     */
    DepartmentId: number;

    /**
     * Name of the department.
     */
    DepartmentName: string = "";

    /**
     * Rank of the department, use it for sorting.
     */
    Rank: number;

    /**
     * Label for the reference range column for this department.
     */
    RangeLabel: string;

    /**
     * Department-level note.
     */
    Note: Note;

    /**
     * List of examination results.
     */
    Results: ExaminationBase[];
  }
}

export { Skyware.Lis.AkitaModel.Results.DepartmentResults };

